core/commands/repo.go

```diff
@@ -152,3 +156,56 @@ RepoPath        string the path to the repo being currently used
 		},
 	},
 }
+
+var repoFsckCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Removes repo lockfiles",
+		ShortDescription: `
+'ipfs repo fsck' is a plumbing command that will remove repo and level db lockfiles.
+`,
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		if !n.OnlineMode() {
```

Comment:
so in this case, we want to make sure that we are *offline*. If we try running a fsk while the daemon is running, we would end up deleting the valid locks that the running daemon is holding